internal/service/s3vectors/testdata/tmpl/vector_bucket_policy_basic.gtpl

```diff
@@ -0,0 +1,29 @@
+resource "aws_s3vectors_vector_bucket" "test" {
+{{- template "region" }}
+  vector_bucket_name = var.rName
+}
+
+resource "aws_s3vectors_vector_bucket_policy" "test" {
+{{- template "region" }}
+  vector_bucket_arn = aws_s3vectors_vector_bucket.test.vector_bucket_arn
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Id": "writePolicy",
+  "Statement": [{
+    "Sid": "writeStatement",
+    "Effect": "Allow",
+    "Principal": {
+      "AWS": "${data.aws_caller_identity.current.account_id}"
+    },
+    "Action": [
+      "s3vectors:PutVectors"
+    ],
+    "Resource": "*"
+  }]
+}
+EOF
+}
+
+data "aws_caller_identity" "current" {}
```

Comment:
Nit, resource type that we're testing should be at the top and there should be a newline at the end of the file